src/charm.py

```diff
@@ -305,13 +300,15 @@ def _reconfigure(self, event) -> None:
                 # compare sets of mongod replica set members and juju hosts
                 # to avoid unnecessary reconfiguration.
                 if replset_members == self.mongodb_config.hosts:
+                    self._set_unit_active_if_needed()
                     return
 
                 # remove members first, it is faster
                 logger.info("Reconfigure replica set")
                 for member in replset_members - self.mongodb_config.hosts:
```

Comment:
Just Q: is `self.mongodb_config.hosts` return a new list of members on the `relation_departed` event on the leader?
I want to avoid the situation when the leader reverts a departed member.

Reply:
Updated the code